packages/kmod-6.1-nvidia-r570/grid-license-check.service

```
@@ -0,0 +1,20 @@
+[Unit]
+Description=GRID License Check
+RefuseManualStart=true
+RefuseManualStop=true
+DefaultDependencies=no
+After=nvidia-gridd.service
+Requires=nvidia-gridd.service
+
+[Service]
+Type=oneshot
+ExecCondition=/usr/bin/ghostdog match-nvidia-driver grid
+ExecStart=-/usr/bin/truncate /tmp/.nvidia-gridd-license
+ExecStart=/usr/bin/nvidia-smi -q
+ExecStart=/usr/bin/grep -q "License Status.*: Licensed" /tmp/.nvidia-gridd-license
+ExecStart=/usr/bin/touch /etc/drivers/.grid-licensed
+ExecStart=/usr/bin/systemctl stop grid-license-check.timer
```

I'd always recommend `--no-block` unless you're very sure this could never possibly deadlock:
```suggestion
ExecStart=/usr/bin/systemctl stop grid-license-check.timer --no-block
```